src/kintone/client.ts

```diff
@@ -61,13 +61,30 @@ const buildHttpsAgent = (options: {
     return new https.Agent({ ...clientAuth });
   }
 
-  const { protocol, hostname, port } = new URL(options.proxy);
-  return HttpsProxyAgent({
+  const { protocol, hostname, port, username, password } = new URL(
+    options.proxy
+  );
+  const proxyOptions: httpsProxyAgent.HttpsProxyAgentOptions = {
     protocol: protocol,
     host: hostname,
     port: port,
     ...clientAuth,
-  });
+  };
+
+  if (username.length > 0 && password.length > 0) {
+    proxyOptions.headers = {
+      "Proxy-Authorization": generateProxyAuthorizationStr(username, password),
```

Comment:
IMO: It can be a more precise name than `Str` like `Value` or `HeaderValue`  

```suggestion
      "Proxy-Authorization": generateProxyAuthorizationHeaderValue(username, password),
```

Another idea:

```suggestion
      "Proxy-Authorization": `Basic ${generateProxyAuthorizationCredentials(username, password)}`
```